src/sql-parser/src/parser.rs

```diff
@@ -2991,6 +2987,18 @@ impl<'a> Parser<'a> {
         }
     }
 
+    fn parse_kw_with_options<T, F>(&mut self, f: F) -> Result<Vec<T>, ParserError>
```

Comment:
I'd love to think of these as just "keyword options". Drop the `WITH`—it's cleaner!

```suggestion
    fn parse_kw_options<T, F>(&mut self, f: F) -> Result<Vec<T>, ParserError>
```